rust/connlib/libs/tunnel/src/device_channel/tun/tun_android.rs

```diff
@@ -90,6 +90,8 @@ impl IfaceStream {
     }
 
     pub fn read(&self, dst: &mut [u8]) -> std::io::Result<usize> {
+        // We don't read(or write) again from the fd because the given fd number might be reclaimed
+        // so this could make an spurious read/write to another fd and we DEFINETLY don't want that
```

Comment:
I think it would make sense to build an abstraction here that only gives you access to the `fd` in a closure after checking that it is no closed. That would reduce this repeated checking down to a single function.